### PR TITLE
added vendor list feature to the inventory items

### LINKED ISSUE #272: itemVendorList doesn't show up in the InventoryItem object
Hi there :)

I'm receiving the following soap response for an inventory item:

```xml
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <soapenv:Header>
    <platformMsgs:documentInfo xmlns:platformMsgs="urn:messages_2016_1.platform.webservices.netsuite.com">
      <platformMsgs:nsId>WEBSERVICEXXXXXX</platformMsgs:nsId>
    </platformMsgs:documentInfo>
  </soapenv:Header>
  <soapenv:Body>
    <getResponse xmlns="urn:messages_2016_1.platform.webservices.netsuite.com">
      <readResponse>
        <platformCore:status xmlns:platformCore="urn:core_2016_1.platform.webservices.netsuite.com" isSuccess="true"/>

<!-- and then vendor information like this: -->

          <listAcct:itemVendorList>
            <listAcct:itemVendor>
              <listAcct:vendor xmlns:platformCore="urn:core_2016_1.platform.webservices.netsuite.com" internalId="20">
                <platformCore:name>Spring Water</platformCore:name>
              </listAcct:vendor>
              <listAcct:purchasePrice>16.14</listAcct:purchasePrice>
              <listAcct:preferredVendor>true</listAcct:preferredVendor>
            </listAcct:itemVendor>
          </listAcct:itemVendorList>

<!-- the rest of the response ... -->

```

but in the response object the mentioned vendor information is missing

### PR DESCRIPTION
**Pull request by rudyyazdi**
this will resolve  #272

### CODE REVIEW

File: lib/netsuite/records/item_vendor_list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module NetSuite
+  module Records
+    class ItemVendorList
```
Comment (iloveitaly) [inline lib/netsuite/records/item_vendor_list.rb]: @rudyyazdi Is there a reason the `Support::Sublist` base class wouldn't work for you here? I'd prefer to use that base class for all sublists if possible.

**Comment by iloveitaly** [comment]
@rudyyazdi Thanks for this contribution!

Would you mind squashing your commits to 2-3 core commits and eliminating the cleanup and merge commits?

**Comment by rudyyazdi** [comment]
@iloveitaly sure man :)

File: lib/netsuite/records/item_vendor_list.rb (hunk 1 above)
Comment (rudyyazdi) [inline lib/netsuite/records/item_vendor_list.rb]: I wasn't aware of `Support::Sublist` I think it makes more sense :)

**Pull request closed without merging**